### DISCUSSION TITLE
Does npm-packager respect peerDependencies versions?

### DISCUSSION THREAD

**Discussion by FreePhoenix888** [deep-foundation/Discussions#18] [Q&A] [answered]
# Example
I have package `@deep-foundation/payments-tinkoff-c2b-test` which has peerDependency: `@deep-foundation/payments-tinkoff-c2b: "^1.0.0"`
What will happen If I have installed `@deep-foundation/payments-tinkoff-c2b@2.0.0` and I will download `@deep-foundation/payments-tinkoff-c2b-test` which requires `@deep-foundation/payments-tinkoff-c2b: "^1.0.0"`?

**Comment by konard** [comment]
@FreePhoenix888 at the moment `npm-packager` uses only regular dependencies, and it does not look at peer dependencies. Why you need it?

**Reply by FreePhoenix888** [reply]
@Konard , I have `@deep-foundation/payments-tinkoff-c2b-test` package that can , for example, test `@deep-foundation/payments-tinkoff-c2b@^1.0.0` packages. How can I define this `requirement`? 

I want a user to be able to test `@deep-foundation/payments-tinkoff-c2b` package without installing a new copy of specific `@deep-foundation/payments-tinkoff-c2b` that will be chosen as regular dependency
Or that will not happen if the package is already installed and the version match?

**Reply by konard** [reply]
@FreePhoenix888 at the moment, there is no way to set that requirement using `npm-packager` logic. You can check for version using `deep.select` when the test will be executed. And you just install these packages separately, but `test` package can use `@deep-foundation/payments-tinkoff-c2b` in code, if that package is installed, if it is not installed just throw an error with good message. At the moment, we can do it like this in current situation.

@ivansglazunov do you think we should support peer dependencies in `npm-packager`?

**Reply by FreePhoenix888** [reply]
> You can check for version using `deep.select` when the test will be executed

I can check version but it a new extra copy of `payments-tinkoff-c2b` will still be installed because it is dependency of `payments-tinkoff-c2b-test`

**Reply by konard** [reply]
@FreePhoenix888 why you need multiple versions of `payments-tinkoff-c2b` in the system? You told me it is peer dependency. Peer dependency is not installed by `npm-packager`. So I don't see how you would be able to install multiple versions of the package using `payments-tinkoff-c2b-test` package.

**Reply by FreePhoenix888** [reply]
> @FreePhoenix888 why you need multiple versions of `payments-tinkoff-c2b` in the system? You told me it is peer dependency. Peer dependency is not installed by `npm-packager`. So I don't see how you would be able to install multiple versions of the package using `payments-tinkoff-c2b-test` package.

I do not need multiple versions of `payments-tinkoff-c2b`. That is exactly the reason why I write this dusscussion. When user download a package and `-test` package I want `-test` to not download a new extra copy of the package

**Reply by konard** [reply, marked as answer]
@FreePhoenix888 to do it, just do not give for `-test` package direct dependency, make it `peerDependency`, `npm-packager` will just ignore all `peerDependencies`.

**Comment by konard** [comment]
> Does npm-packager respect peerDependencies versions? 

No, it does not.

**Comment by vag1ffm** [comment]
No, it does not.
